src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowDelete.cs

```diff
@@ -74,6 +100,35 @@ public override DbCommand GetCommand(DbConnection connection)
             return command;
         }
 
+        /// <summary>
+        /// Runs a query using the where clause to determine if duplicates are found (causes issues when deleting).
+        /// </summary>
+        private bool checkWhereDuplicate(WhereClause where, string input, DbConnection connection)
+        {
+            using (DbCommand command = connection.CreateCommand())
+            {
+                command.CommandText = input;
+                command.Parameters.AddRange(where.Parameters.ToArray());
+                using (DbDataReader reader = command.ExecuteReader())
+                {
+                    try
+                    {
+                        while (reader.Read())
+                        {
+                            if (reader.GetInt32(0) != 1)
```

Comment:
it looks like if there are multiple rows and any of them contain a 1 this method will return true.  I'd recommend breaking or returning false, if this condition returns false to make the intent of the code more explicit.

Reply:
this is for running the select query on the rows, since GetScalar is not recognized by the tests, I have to use GetInt32 to handle it

Reply:
What do you mean "not recognized"? And by GetScalar do you mean ExecuteScalar? If a test is using this test object https://github.com/microsoft/sqltoolsservice/blob/master/test/Microsoft.SqlTools.ServiceLayer.UnitTests/Utility/TestObjects.cs#L163 then yes, it looks like another case of that feature not being implemented. If that's the case you should implement that instead of using a less-ideal approach in the production code.